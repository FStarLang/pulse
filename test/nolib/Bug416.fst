module Bug416

#lang-pulse

open Pulse.Nolib

assume val ref : int -> prop
assume val foo : x:int{ref x} -> slprop

[@@expect_failure [19]]
ghost
fn test (_ : squash (ref 1))
  requires foo 1
  ensures          foo 2
{
  admit ()
}
