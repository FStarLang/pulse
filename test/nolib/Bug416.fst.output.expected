* Info at Bug416.fst(14,19-14,22):
  - Expected failure:
  - Ill-typed term: Bug416.foo 2
  - Expected a term of type slprop
  - Assertion failed
  - The SMT solver could not prove the query. Use --query_stats for more
    details.
  - See also Bug416.fst(8,23-8,28)

